client/src/api/v2/camera.ts

```diff
@@ -0,0 +1,298 @@
+ */
+export function getPrettyDeviceName(device: CameraDevice) {
+  return device === 'primary' ? 'Primary' : 'Secondary';
+}
+
+/**
+ * `T` Type of the socket.io payload
+ * `U` Type to be returned by the handler
+ */
+interface StatusPayloadOptions<T, U> {
+  /** Initial value for the payload */
+  initValue: T | null;
+  /** Handler for updating payload */
+  payloadHandler?: (
+    setter: React.Dispatch<React.SetStateAction<T | null>>,
+    newPayload: T | null,
+    device: CameraDevice,
+  ) => T | void;
+  /** Handler for return value */
+  returnHandler?: (payload: T | null, device?: CameraDevice) => U | null;
+}
+
+/**
+ * Create hooks for getting payloads
+ *
+ * @param sub Subcomponent
+ * @param shape Shape of received payloads
+ * @param opts Options
+ * @returns Hook for getting a payload
+ */
+function createStatusPayloadHook<T>(
+  sub: string,
+  shape: Runtype<T>,
+  {
+    initValue = null,
+    payloadHandler = (
+      setter: React.Dispatch<React.SetStateAction<T | null>>,
+      newPayload: T | null,
+    ) => setter(newPayload ?? initValue),
+    returnHandler = (payload: T | null) => payload,
+  }: StatusPayloadOptions<T, T>,
+) {
+  return function _hook(device: CameraDevice) {
+    // Only run init once per render
+    useEffect(() => {
+      initStatus(sub, device);
+    }, [device]);
+
+    const [payload, setPayload] = useState(initValue);
+    useChannelShaped(`status-camera-${sub}`, shape, (newPayload: T) =>
+      payloadHandler(setPayload, newPayload, device),
+    );
+
+    return returnHandler(payload, device);
+  };
+}
+
+const CameraRecordingStatusPayloads = Partial({
+  primary: CameraRecordingStatusPayload,
+  secondary: CameraRecordingStatusPayload,
+});
+
+/**
+ * Returns the last status payload published
+ *
+ * @param device Device
+ * @returns Recording status
+ */
+export const useCameraRecordingStatus = createStatusPayloadHook(
+  'recording',
+  CameraRecordingStatusPayloads,
+  {
+    initValue: null,
+  },
+);
+
+const VideoFeedStatus = Record({
+  /** Whether video feed is on/off */
+  online: Boolean,
+});
+type VideoFeedStatus = Static<typeof VideoFeedStatus>;
+
+const VideoFeedStatuses = Partial({
+  primary: VideoFeedStatus,
+  secondary: VideoFeedStatus,
+}).Or(Null);
+
+/**
+ * Returns the last received status of the video feeds from `/v3/camera/video-feed/status/<primary/secondary>`
+ *
+ * @returns A VideoFeedStatus for each device
+ */
+export const useVideoFeedStatus = createStatusPayloadHook(
+  'video-feed',
+  VideoFeedStatuses,
+  { initValue: {} },
+);
+
+const CameraStatus = Record({
+  /** Whether camera is connected / not connected */
+  connected: Boolean,
+});
+
+/**
+ * Returns the last received connection status of the camera client to the mqtt broker
+ *
+ * @param device Device
+ * @returns Camera status
+ */
+export const useCameraStatus = (device: CameraDevice) =>
+  createStatusPayloadHook(device, CameraStatus, {
+    initValue: { connected: false },
```

Comment:
Same here, could you remind me why `payloadHandler` was removed? [From here](https://github.com/monash-human-power/dashboard/blob/master/client/src/api/v2/camera.js#L234)

Reply:
Looks like I'd made the hook return the entire payload making the handler redundant as the default implementation is suitable. (I'm imagining publishing the camera's IP in the same status payload - so it makes sense to return the whole object even if it is just a boolean for the time being)